bpf/kmesh/workload/sockops_tuple.c

```diff
@@ -14,102 +14,156 @@
-        key->ipv4.daddr = skops->local_ip4;
-        key->ipv4.dport = bpf_htonl(skops->local_port) >> 16;
-    }
+	struct kmesh_manager_key key = {0};
+	key.ipaddress = ip;
+	__u8 *value = bpf_map_lookup_elem(&map_of_kmesh_manager, &key);
+	if (value)
+		return 1;
+	return 0;
 }
 
 // clean map_of_auth
 static inline void clean_auth_map(struct bpf_sock_ops *skops)
 {
-    struct bpf_sock_tuple tuple_key = {0};
-    build_auth_key(skops, &tuple_key);
-    long ret = bpf_map_delete_elem(&map_of_auth, &tuple_key);
-    if(ret && ret != -ENOENT) {
-        BPF_LOG(INFO, SOCKOPS, "map_of_auth bpf_map_delete_elem failed, ret: %d\n", ret);
-    }
+	struct bpf_sock_tuple tuple_key = {0};
+	// auth run PASSIVE ESTABLISHED CB now. In thie state cb
+	// tuple info src is server info, dst is client info
+	// During the auth, src must set the client info and dst set
+	// the server info when we transmitted to the kmesh auth info.
+	// In this way, auth can be performed normally.
+	tuple_key.ipv4.saddr = skops->remote_ip4;
+	tuple_key.ipv4.daddr = skops->local_ip4;
+	tuple_key.ipv4.sport = skops->remote_port >> 16;
+	tuple_key.ipv4.dport = bpf_htonl(skops->local_port) >> 16;
+	long ret = bpf_map_delete_elem(&map_of_auth, &tuple_key);
+	if(ret && ret != -ENOENT)
+		BPF_LOG(INFO, SOCKOPS, "map_of_auth bpf_map_delete_elem failed, ret: %d\n", ret);
 }
 
 // insert an IPv4 tuple into the ringbuf
-static inline void insert_ipv4_tuple(struct bpf_sock_ops *skops)
+static inline void auth_ip_tuple(struct bpf_sock_ops *skops)
 {
-    if (skops->family == AF_INET) {
-        struct ringbuf_msg_type *msg = bpf_ringbuf_reserve(&map_of_tuple, sizeof(*msg), 0);
-        if (!msg) {
-            BPF_LOG(WARN, SOCKOPS, "can not alloc new ringbuf in map_of_tuple");
-            return;
-        }
-        (*msg).tuple.ipv4.daddr = skops->local_ip4;
-        (*msg).tuple.ipv4.saddr = skops->remote_ip4;
-        // local_port is host byteorder
-        (*msg).tuple.ipv4.dport = bpf_htonl(skops->local_port) >> 16;
-        // remote_port is network byteorder
-        (*msg).tuple.ipv4.sport = skops->remote_port >> 16;
-        (*msg).type = (__u32)IPV4;
-        bpf_ringbuf_submit(msg, 0);
-    }
+	struct ringbuf_msg_type *msg = bpf_ringbuf_reserve(&map_of_tuple, sizeof(*msg), 0);
+	if (!msg) {
+		BPF_LOG(WARN, SOCKOPS, "can not alloc new ringbuf in map_of_tuple");
+		return;
+	}
+	// auth run PASSIVE ESTABLISHED CB now. In thie state cb
+	// tuple info src is server info, dst is client info
+	// During the auth, src must set the client info and dst set
+	// the server info when we transmitted to the kmesh auth info.
+	// In this way, auth can be performed normally.
+	(*msg).tuple.ipv4.saddr = skops->remote_ip4;
+	(*msg).tuple.ipv4.daddr = skops->local_ip4;
+	// remote_port is network byteorder
+	(*msg).tuple.ipv4.sport = skops->remote_port >> 16;
+	// local_port is host byteorder
+	(*msg).tuple.ipv4.dport = bpf_htonl(skops->local_port) >> 16;
```

Comment:
Isn't this same with that in clean_auth_map

Reply:
Yes, the logic is the same. Let's make a simple macro wrapper.